File: vanillaWebSetup/.vanillaWebSetup.py
import os
import sys

here = os.getcwd()

NAME = sys.argv[1]
ARG_COUNT = len(sys.argv)

if (ARG_COUNT != 2):
    print('------------------------------------------')
    print('You must pass 1 argument into this script.')
    print('------------------------------------------')

else:
    NAME = sys.argv[1]
    files = ['index.html', 'style.css', 'app.js']
    starts_stops = [['<!doctype html>', '</html>'],
                    ['*,', '    min-width: 100vw; }'],
                    ['Testing Start', 'Testing Stop']]

    def reader(file, lines_arr, good_text):
        start = good_text.index(lines_arr[0])
        stop = (good_text.index(lines_arr[1]) + 1)

        with open(file, 'w') as new_file:
            for i in range(start, stop):
                if good_text[i] == '  <title>The HTML5 Herald</title>':
                    new_file.write(f'<title>{NAME}</title>')
                else:
                    new_file.write(good_text[i])

    with open('./resets.txt', 'r') as raw_text:
        text = raw_text.read().split('\n')
        print(text)
        for file in files:
            reader(file, starts_stops[files.index(file)], text)

File: vanillaWebSetup/.vanillaWebSetup.js
const path = require('path');
const fs = require('fs');
const here = path.resolve();

const NAME = process.argv[2];
const ARG_COUNT = process.argv.length;

const htmlStart = '<!doctype html>';
const htmlEnd = '</html>';
const cssStart = '*,';
const cssEnd = 'min-width: 100vw; }';

const reader = (start, stop, text) => {
    const textArr = text.split('\n');
    const beginning = textArr.indexOf(start);
    const end = textArr.indexOf(stop);

    return textArr
        .slice(beginning, end)
        .toString()
        .replace(/,/g, '\n')
        .replace(/The HTML5 Herald/g, NAME);
};

if (ARG_COUNT <= 2) {
    console.log(`
    ------------------------------------------
    You must pass 1 argument into this script.
    ------------------------------------------
    `);
} else if (ARG_COUNT === 3) {
    fs.readFile('./resets.txt', 'utf-8', (err, text) => {
        if (err) {
            console.error(err);
        } else {
            const files = ['index.html', 'style.css', 'app.js'];
            const theGoodStuff = [
                reader(htmlStart, htmlEnd, text),
                reader(cssStart, cssEnd, text),
                'Testing'
            ];

            files.forEach(file =>
                fs.writeFile(file, theGoodStuff[files.indexOf(file)], err =>
                    err
                        ? console.error(err)
                        : console.log(`${file} was created in ${here}`)
                )
            );
        }
    });
} else {
    console.log(`
    ------------------------------------
    This script only accepts 1 argument.
    ------------------------------------
    `);
}
